components/engine_rocks/src/perf_context_impl.rs

```diff
@@ -102,25 +196,207 @@ impl PerfContextStatistics {
         }
         let mut ctx = RawPerfContext::get();
         ctx.reset();
-        self.apply_write_perf_settings();
-        self.write_wal_time = 0;
-        self.pre_and_post_process = 0;
-        self.db_mutex_lock_nanos = 0;
-        self.write_thread_wait = 0;
-        self.write_memtable_time = 0;
-        self.write_scheduling_flushes_compactions_time = 0;
-        self.db_condition_wait_nanos = 0;
-        self.write_delay_time = 0;
+        self.apply_perf_settings();
     }
 
     pub fn report(&mut self) {
         match self.kind {
             PerfContextKind::RaftstoreApply => {
-                report_perf_context!(self, APPLY_PERF_CONTEXT_TIME_HISTOGRAM_STATIC);
+                report_write_perf_context!(self, APPLY_PERF_CONTEXT_TIME_HISTOGRAM_STATIC);
             }
             PerfContextKind::RaftstoreStore => {
-                report_perf_context!(self, STORE_PERF_CONTEXT_TIME_HISTOGRAM_STATIC);
+                report_write_perf_context!(self, STORE_PERF_CONTEXT_TIME_HISTOGRAM_STATIC);
             }
+            PerfContextKind::GenericRead => {
+                // TODO: Currently, metrics about reading is reported in other ways.
+                // It is better to unify how to report the perf metrics.
+                //
+                // Here we only record the PerfContext data into the fields.
+                self.read = ReadPerfContext::capture();
+            }
```

Comment:
make it generics?

Reply:
Sorry, what do you want it to be generic about? This is under `engine_rocks`, so it needn't be generic over engine.

Reply:
I hadn't thought clearly. Never mind...

So we will report these statistics as metrics in the future?

Reply:
And those statistics will also return as coprocessor response...

I hope that there will be an explicit way to remind people of their multiple purposes.

Reply:
We already do. Now, it's in the coprocessor mod where we report these statistics.
But I think we can report these metrics like raftstore statistics here as well. So,  it will look more consistent about reporting rocksdb statistics.

Reply:
What about changing `report_metrics` to `stop_observe` and add another trait function to `PerfContext` to get the inner data?